docs/documentation/development/guidelines.md

```diff
@@ -22,6 +22,7 @@ Note that this list also serves as a checklist for code reviews on pull requests
 1. Generics must be used where applicable.
 1. Code should not show any warnings. Warnings that cannot be circumvented should be suppressed by using the @SuppressWarnings annotation. 
 1. For dependency injection, OSGi Declarative Services should be used.
+1. OSGi Declarative Services should be declared using annotations. The IDE will take care of the service *.xml file creation. See the binding archetype for an [example here](https://github.com/eclipse/smarthome/blob/master/tools/archetype/binding/src/main/resources/archetype-resources/src/main/java/internal/__bindingIdCamelCase__HandlerFactory.java#L32) or a more advanced configuration [with injection here](https://github.com/eclipse/smarthome/blob/master/bundles/config/org.eclipse.smarthome.config.xml/src/main/java/org/eclipse/smarthome/config/xml/ConfigXmlConfigDescriptionProvider.java#L34) and [here](https://github.com/eclipse/smarthome/blob/master/bundles/config/org.eclipse.smarthome.config.xml/src/main/java/org/eclipse/smarthome/config/xml/ConfigXmlConfigDescriptionProvider.java#L59).
```

Comment:
I would refrain from linking to specific code lines on the master branch - that can change too often. And the guidelines are also re-used to the openHAB docs. I'd suggest to link to the official docs at http://enroute.osgi.org/services/org.osgi.service.component.html instead.